service/src/main/java/aero/minova/cas/service/QueueService.java

```diff
@@ -0,0 +1,366 @@
+		SecurityContextHolder.getContext().setAuthentication(new Authentication() {
+
+			@Override
+			public String getName() {
+				return "CAS QueueService";
+			}
+
+			@Override
+			public void setAuthenticated(boolean isAuthenticated) throws IllegalArgumentException {
+
+			}
+
+			@Override
+			public boolean isAuthenticated() {
+				return true;
+			}
+
+			@Override
+			public Object getPrincipal() {
+				return null;
+			}
+
+			@Override
+			public Object getDetails() {
+				return null;
+			}
+
+			@Override
+			public Object getCredentials() {
+				return null;
+			}
+
+			@Override
+			public Collection<? extends GrantedAuthority> getAuthorities() {
+				return null;
+			}
+		});
+
+		// Holt sich alle Nachrichten, die noch nicht versandt wurden.
+		Table messagesToBeSend = getNextMessage();
+
+		if (messagesToBeSend != null && messagesToBeSend.getRows().size() > 0) {
+
+			for (Row pendingMessage : messagesToBeSend.getRows()) {
+
+				// Falls die Nachricht älter ist als das allowedMessageAge und falls die NumberOfAttempts höher ist als die allowedNumberOfAttempts, muss die
+				// Nachricht gelöscht werden.
+				if (pendingMessage.getValues().get(8).getInstantValue().isBefore(Instant.now().minus(allowedMessageAge, ChronoUnit.DAYS))
+						|| pendingMessage.getValues().get(7).getIntegerValue() >= allowedNumberOfAttempts) {
+					deleteMessage(pendingMessage);
+					continue;
+				}
+
+				// Versuche die Nachricht an den Dienst zu verschicken.
+				boolean sendSuccessfull = sendMessage(pendingMessage);
+				if (sendSuccessfull) {
+					safeAsSent(pendingMessage);
+				} else {
+					logger.logNewsfeed(pendingMessage.getValues().get(1).getStringValue() + " is not reachable!");
+				}
+				increaseAttempts(pendingMessage);
+			}
+		}
+	}
+
+	/**
+	 * Speichert eine Nachricht in der Datenbank, welche beim nächsten Intervall verschickt wird.
+	 */
+	@Override
+	public void accept(Object t, Object u) {
+
+		if (t instanceof Table && u instanceof ResponseEntity) {
+			Table table = (Table) t;
+
+			Map<String, BiFunction> applyableFunctions = serviceMessageCreators.get(table.getName());
+
+			if (applyableFunctions != null) {
+
+				// Wenn eine Prozedur ausgeführt wurde, müssen Nachrichten für alle betroffenen Dienste generiert werden.
+				for (Map.Entry<String, BiFunction> entry : applyableFunctions.entrySet()) {
+					@SuppressWarnings("unchecked")
+					String message = (String) entry.getValue().apply(t, u);
+					saveMessage(message, table.getName(), entry.getKey());
+				}
+			}
+		}
+	}
+
+	/**
+	 * Speichert eine Nachricht für einen Service.
+	 * 
+	 * @param message
+	 *            Die Nachricht, die gespeichert werden soll.
+	 * @param procedureName
+	 *            Die Prozedur, wegen welcher die Nachricht erstellt wurde.
+	 * @param serviceName
+	 *            Der Name des Services, an welchen die Nachricht geschickt werden soll.
+	 */
+	private void saveMessage(String message, String procedureName, String serviceName) {
```

Comment:
Hier müsste man entsprechend als Argumente message und topic haben.
Die SQL-Prozedur würde dann entsprechende für jeden Dienst des Topics einen Eintrag in die messages-Tabelle machen.